### PR TITLE
Added integration tests for registry

### PR DESCRIPTION
**Pull request by castlemilk**
Adding integration tests for the registry addon as mentioned in #4637. Validates the following functions:

- registry is accessible from within the cluster (via an `busybox` pod)
- registry is accessible from outside the cluster (via newly added `registry-proxy`)

This aims to prevent breaking changes in the future with regards to modifying the registry addon configuration and functionality.
This PR simply uses curl to validate connectivity is functional, it may be worthwhile actually pushing/pulling images from the registry from the different ingress mechanisms

@medyagh

### CODE REVIEW

**Comment by medyagh** [comment]
Unit tests failed?

**Comment by castlemilk** [comment]
@medyagh resolved linting issue.

**Comment by medyagh** [comment]
@minikube-bot OK to test

File: test/integration/addons_test.go [hunk 1]
```diff
@@ -190,7 +190,57 @@ func testServicesList(t *testing.T) {
...
+
+	if err := util.WaitForDockerRegistryRunning(t); err != nil {
+		t.Fatalf("waiting for registry to be up: %v", err)
+	}
+
+	// Check access from outside the cluster on port 5000, validing connectivity via registry-proxy
+	checkExternalAccess := func() error {
+		t.Log("checking registry access from outside cluster")
+		expectedStr := "200"
+		runCmd := fmt.Sprintf("curl -sS -o /dev/null -w '%%{http_code}' http://127.0.0.1:5000")
```
Comment (medyagh) [inline test/integration/addons_test.go]: Please use the retryable http instead of curl, not every system has curl and it is not rertiable 

please change to be like this one
https://github.com/kubernetes/minikube/blob/master/test/integration/proxy_test.go#L153

or this one https://github.com/kubernetes/minikube/blob/e236c966a74d7952b9940e2f7bac1ee1fe38239a/test/integration/version_upgrade_test.go#L38

**Comment by RA489** [comment]
@minikube-bot OK to test

**Comment by castlemilk** [comment]
@minikube-bot OK to test

File: test/integration/addons_test.go [hunk 2]
```diff
@@ -190,7 +190,74 @@ func testServicesList(t *testing.T) {
...
+		registryEndpoint := "http://" + strings.TrimSpace(s) + ":5000"
+		u, err := url.Parse(registryEndpoint)
+
+		if err != nil {
+			t.Fatalf("failed to parse %q: %v", s, err)
+		}
+
+		resp, err := retryablehttp.Get(u.String())
+		if err != nil {
+			t.Fatalf("failed get: %v", err)
```
Comment (medyagh) [inline test/integration/addons_test.go]: the t.Fatalf will make the test fail, and will not give it a chace to be retried by util.Retry.
you could either return error or not do the util.Retry at all.

File: pkg/minikube/extract/testdata/test.json [hunk 3]
```diff
@@ -1,4 +1,7 @@
 {
+	"Hint: This is not a URL, come on.": "",
```
Comment (tstromberg) [inline pkg/minikube/extract/testdata/test.json]: Please revert this unrelated change. It's not your fault though - it's due to one of our new localization tests. Thanks!

**Comment by tstromberg** [comment]
PR looks great. Will be happy to merge it once comment is resolved.

**Comment by castlemilk** [comment]
@tstromberg Done! 🎉

**Comment by tstromberg** [comment]
@minikube-bot OK to test

**Comment by tstromberg** [comment]
Thank you!
